test/subscription_map.hpp

```diff
@@ -439,31 +432,48 @@ class multiple_subscription_map
     // Insert a key => value at the specified subscription path
     // returns the handle and true if key was inserted, false if key was updated
     template <typename V>
-    std::pair<handle, bool> insert_or_update(MQTT_NS::string_view subscription, Key const &key, V&& value) {
+    std::pair<handle, bool> insert_or_update(MQTT_NS::string_view subscription, Key&& key, V&& value) {
```

Comment:
```suggestion
    template <typename K, typename V>
    std::pair<handle, bool> insert_or_update(MQTT_NS::string_view subscription, K&& key, V&& value) {
```

You are misusing class template parameter `Key` as forwarding reference. `Key&&` is NOT forwarding reference. It is rvalue reference because it is not deduced.

I guess that is the reason you got compile errer at https://github.com/redboltz/mqtt_cpp/pull/713/files#r512463471

You also need to replace `std::forward<Key>(key)` with `std::forward<K>(key)`.

Reply:
Yes, the additional insert_or_update can be removed now